scripts/src/public/services/orders.ts

```diff
@@ -550,3 +550,43 @@ export async function createOutgoingTransferOrder(recipientWalletAddress: string
 
 	return openOrderDbToApi(order, sender.id);
 }
+
+export async function createIncomingTransferOrder(title: string, description: string, memo: string, senderWalletAddress: string, appId: string, receiver: User, receiverDeviceId: string): Promise<Order> {
+	logger().info("creating an incoming transfer order");
+	const receiverWallet = (await receiver.getWallets(receiverDeviceId)).lastUsed();
+	if (!receiverWallet) {
+		throw UserHasNoWallet(receiver.id, receiverDeviceId);
+	}
+
+	const order = db.IncomingTransferOrder.new({
+		offerId: `cross-app_${appId}_to_${receiver.appId}`,
+		status: "pending",
+		amount: 0, // not allowed to insert an order without amount
+		blockchainData: {
+			sender_address: senderWalletAddress,
+			recipient_address: receiverWallet.address,
+			memo
+		}
+	}, {
+		user: receiver,
+		type: "earn",
+		wallet: receiverWallet.address,
+		meta: { title, description }
+	});
+
+	await order.save();
+
+	// we create a redis entry so this memo can uniquely identify this order.id down the line
+	// i.e. when the watch we gonna add triggers a callback
+	const parts = memo.split("-"); // extract order id from memo
+	const redis = getRedisClient();
+	await redis.async.set(`transfer:${parts[2]}`, order.id);
+
+	// adding a watch
+	const res = await setWatcherEndpoint([receiverWallet.address], receiver.appId);
```

Comment:
You shouldn't call this here. Only a call to `addWatcherEndpoint`. 
The `setWatcherEndpoint` should only be called from the initialization code of the marketplace-internal.

Reply:
elaborate on this one. I see it's called once from `initPaymentCallbacks` that is invoked on app startup, but I don't know the address to watch then. It is created dynamically.

Reply:
1. The addresses passed in `setWatcherEndpoint` are supposed to be permanent address that you will always watch.
2. calling `setWatcherEndpoint` creates or overrides these permanent addresses and and sets the webhook to cal back when payments are found. 
3. When you want to watch an address for a short period of time like in p2p or in this incoming transfer, use `addWatcherEndpoint`. I guess naming could be better.

Reply:
done